dandiapi/api/storage.py

```diff
@@ -1,13 +1,66 @@
-from typing import Any
+from __future__ import annotations
+
+from datetime import timedelta
+import hashlib
+from typing import Any, Iterator
 from urllib.parse import urlsplit, urlunsplit
 
+import boto3
+from botocore.exceptions import ClientError
+from dandischema.digests.dandietag import PartGenerator
 from django.conf import settings
 from django.core.files.storage import Storage, get_storage_class
+from minio.error import NoSuchKey
 from minio_storage.policy import Policy
 from minio_storage.storage import MinioStorage, create_minio_client_from_settings
+from s3_file_field._multipart_boto3 import Boto3MultipartManager
+from s3_file_field._multipart_minio import MinioMultipartManager
 from storages.backends.s3boto3 import S3Boto3Storage
 
 
+class ChecksumCalculatorFile:
+    """File-like object that calculates the checksum of everything written to it."""
+
+    def __init__(self):
+        self.h = hashlib.sha256()
+
+    def write(self, bytes):
+        self.h.update(bytes)
+
+    @property
+    def checksum(self):
+        return self.h.hexdigest()
+
+
+class DandiMultipartMixin:
+    @staticmethod
+    def _iter_part_sizes(file_size: int, part_size: int = None) -> Iterator[tuple[int, int]]:
```

Comment:
```suggestion
    def _iter_part_sizes(file_size: int, part_size: int | None = None) -> Iterator[tuple[int, int]]:
```

Reply:
I did you one better and removed the parameter. :)